src/libraries/Microsoft.PowerFx.Core/Public/Config/ReadOnlySymbolTable.cs

```diff
@@ -345,6 +345,28 @@ public IEnumerable<NamedFormulaType> SymbolNames
             }
         }
 
+        /// <summary>
+        /// Returns true if the symbol is defined. 
+        /// This includes both variables (that would be found with <see cref="TryLookupSlot(string, out ISymbolSlot)"/>, as well as constants (like Enums, OptionSets, constants).
+        /// It does not include Functions since those are in a different namespace. 
+        /// </summary>
+        /// <param name="name">logical or display name of a symbol.</param>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public bool TryGetType(string name, out FormulaType type)

```

Comment:
Due to another bug, this will not give right types for enums for option sets  - see  #2342
But you'd have the same problem if you enumerated the symbols and got the types that way.